src/Polynomials.jl

```diff
@@ -363,6 +365,7 @@ p(4)   # 57 = 1 + 2*4 + 3*4^2
 """
 function polyval{T,S}(p::Poly{T}, x::S)
     R = promote_type(T,S)
+    T<:AbstractFloat && any(isnan(p.a)) && convert(R, NaN)
```

Comment:
Hmm, I definitely think we should have such a check, as it allows for more general definitions of polynomials. (Part of why I rewrote the display code was to have more general coefficients.) Is the problem for you the ugliness of the construct, or is it something else?